src/cpu_depth_packet_processor.cpp

```diff
@@ -332,63 +326,26 @@ class CpuDepthPacketProcessorImpl: public WithPerfLogging
 
   int32_t decodePixelMeasurement(unsigned char* data, int sub, int x, int y)
   {
+    if (x < 1 || y < 0 || 510 < x || 423 < y)
+    {
+      return lut11to16[0];
+    }
+
+    int r1zi = (x >> 2) + ((x & 0x3) << 7); // range 1..510
+    r1zi = r1zi * 11L; // range 11..5610
+
     // 298496 = 512 * 424 * 11 / 8 = number of bytes per sub image
     uint16_t *ptr = reinterpret_cast<uint16_t *>(data + 298496 * sub);
     int i = y < 212 ? y + 212 : 423 - y;
     ptr += 352*i;
 
-    /**
-     r1.yz = r2.xxyx < l(0, 1, 0, 0) // ilt
-     r1.y = r1.z | r1.y // or
-     */
-    bool r1y = x < 1 || y < 0;
-    /*
-    r1.zw = l(0, 0, 510, 423) < r2.xxxy // ilt
-    r1.z = r1.w | r1.z // or
-    */
-    bool r1z = 510 < x || 423 < y;
-    /*
-    r1.y = r1.z | r1.y // or
-    */
-    r1y = r1y || r1z;
-    /*
-    r1.y = r1.y & l(0x1fffffff) // and
-    */
-    int r1yi = r1y ? 0xffffffff : 0x0;
-    r1yi &= 0x1fffffff;
-
-    /*
-    bfi r1.z, l(2), l(7), r2.x, l(0)
-    ushr r1.w, r2.x, l(2)
-    r1.z = r1.w + r1.z // iadd
-    */
-    int r1zi = bfi(2, 7, x, 0);
-    int r1wi = x >> 2;
-    r1zi = r1wi + r1zi;
-
-    /*
-    imul null, r1.z, r1.z, l(11)
-    ushr r1.w, r1.z, l(4)
-    r1.y = r1.w + r1.y // iadd
-    r1.w = r1.y + l(1) // iadd
-    r1.z = r1.z & l(15) // and
-    r4.w = -r1.z + l(16) // iadd
-     */
-    r1zi = (r1zi * 11L) & 0xffffffff;
-    r1wi = r1zi >> 4;
-    r1yi = r1yi + r1wi;
+    int r1yi = r1zi >> 4; // range 0..350
     r1zi = r1zi & 15;
-    int r4wi = -r1zi + 16;
-
-    if(r1yi > 352)
-    {
-      return lut11to16[0];
-    }
 
     int i1 = ptr[r1yi];
     int i2 = ptr[r1yi + 1];
```

Comment:
I see no reason to discard `x == 0` in this function.

Python computed what happens for x = 511:
````python
>>> x = 511
>>> r1zi = (x >> 2) + ((x & 0x3) << 7)
>>> r1zi = r1zi * 11
>>> r1yi = r1zi >> 4
>>> r1yi
351
>>> r1zi = r1zi & 15
>>> r1zi
5
````
The `ptr[r1yi + 1]` gets left-shifted 11 positions, and is then discarded due to `& 2047`:
````python
>>> i1 =  0       # Real data 0
>>> i2 = 0xffff  # Garbage
>>> i1 = i1 >> r1zi
>>> i2 = i2 << (16 - r1zi)
>>> (i1 | i2) & 2047
0   # No garbage bits
````
It seems `x == 511` can be allowed here.

The filter stages that follow copy the edge pixels. Allowing the `x == 0 || x == 511` here means the next-to-edge pixels get a slightly different value in filterPixelstage1